### PR TITLE
Fix #56: Add load_balancing create/get/delete pool endpoints

### LINKED ISSUE #56: Add load_balancing create/get/delete pool endpoints

### PR DESCRIPTION
**Pull request by adamchalmers**
I also added a new binary that does end-to-end tests (requires a zone on Cloudflare with all the permissions you want to test).

### CODE REVIEW

File: src/endpoints/load_balancing/mod.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    pub enabled: bool,
+    /// The weight of this origin relative to other origins in the Pool. Based on the configured
+    /// weight the total traffic is distributed among origins within the Pool.
+    pub weight: f64,
+}
+
+// f64 doesn't impl Eq or Hash, so we need some custom implementations.
+impl PartialEq for Origin {
+    fn eq(&self, other: &Self) -> bool {
+        let diff_is_small = (self.weight - other.weight).abs() < 0.01;
```
Comment (gabbifish) [inline src/endpoints/load_balancing/mod.rs]: Nice :)

File: example/main.rs [hunk 2]
```diff
@@ -189,7 +188,11 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         panic!("Either API token or API key + email pair must be provided")
     };
 
-    let api_client = HttpApiClient::new(credentials, HttpApiClientConfig::default(), Environment::Production)?;
+    let api_client = HttpApiClient::new(
```
[the remark this answers is not in the document]
Comment (adamchalmers) [reply example/main.rs]: I have run `cargo fmt`, no changes! Maybe the previous commit hadn't run `cargo fmt`?

File: e2e_tests/main.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
+use clap::{App, AppSettings, Arg};
```
[the remark this answers is not in the document]
Comment (gabbifish) [reply e2e_tests/main.rs]: This makes sense :)

File: example/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabbifish) [reply example/main.rs]: that must have been the case!

File: src/endpoints/load_balancing/pool_details.rs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+use crate::endpoints::load_balancing::Pool;
+use crate::framework::endpoint::{Endpoint, Method};
+
+/// Pool Details
+/// https://api.cloudflare.com/#account-load-balancer-pools-pool-details
+pub struct PoolDetails<'a> {
+    /// The Cloudflare account to create this Pool under.
```
Comment (nickvollmar) [inline src/endpoints/load_balancing/pool_details.rs]: I don't think this comment is accurate here –– maybe "The Cloudflare account of this pool"?

File: src/endpoints/load_balancing/delete_pool.rs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+use crate::framework::endpoint::{Endpoint, Method};
+use crate::framework::response::ApiResult;
+
+/// Delete Pool
+/// https://api.cloudflare.com/#account-load-balancer-pools-delete-pool
+pub struct DeletePool<'a> {
+    /// The Cloudflare account to create this Pool under.
+    pub account_identifier: &'a str,
+    /// Which pool to retrieve the details of.
+    pub identifier: &'a str,
```
Comment (nickvollmar) [inline src/endpoints/load_balancing/delete_pool.rs]: The parameter comments should be updated to reflect the semantics of DeletePool –– e.g. "Which pool to delete"

File: src/endpoints/load_balancing/pool_details.rs (hunk 4 above)
Comment (adamchalmers) [reply src/endpoints/load_balancing/pool_details.rs]: Good points, fixed!

File: src/endpoints/load_balancing/delete_pool.rs (hunk 5 above)
Comment (adamchalmers) [reply src/endpoints/load_balancing/delete_pool.rs]: Thanks, fixed

**Pull request merged**